kaldb/src/main/java/com/slack/kaldb/logstore/search/LogIndexSearcherImpl.java

```diff
@@ -68,15 +60,24 @@ public static SearcherManager searcherManagerFromPath(Path path) throws IOExcept
 
   public LogIndexSearcherImpl(
       SearcherManager searcherManager, ConcurrentHashMap<String, LuceneFieldDef> chunkSchema) {
+    this.openSearchAdapter = new OpenSearchAdapter(chunkSchema);
+    this.refreshListener =
+        new ReferenceManager.RefreshListener() {
+          @Override
+          public void beforeRefresh() {
+            // no-op
+          }
+
+          @Override
+          public void afterRefresh(boolean didRefresh) {
+            openSearchAdapter.reloadSchema();
+          }
+        };
     this.searcherManager = searcherManager;
-    this.analyzer = new StandardAnalyzer();
-    this.chunkSchema = chunkSchema;
-    this.openSearchAggregationAdapter = new OpenSearchAggregationAdapter(chunkSchema);
-  }
+    this.searcherManager.addListener(refreshListener);
 
-  // Lucene's query parsers are not thread safe. So, create a new one for every request.
-  private QueryParser buildQueryParser() {
-    return new KaldbQueryParser(SystemField.ALL.fieldName, analyzer, chunkSchema);
+    // initialize the adapter with whatever the default schema is
+    openSearchAdapter.reloadSchema();
```

Comment:
can this just be the default before in the open search adapter constructor?

Reply:
I had this approach initially, but I didn't like mixing the patterns where the consumer of the `OpenSearchAdapter ` class was expected to handle reloading themselves, but the first time was "automatic". I think it's much cleaner when the consumer is entirely responsible for this. It also allows for scenarios where you initialize the class, load the data, then initialize the schema without invoking an extra schema reload.